### PR TITLE
fgdbg client project structure + simple server connection

### PR DESCRIPTION
**Pull request by feresr**
I've decided to use [compose-mpp](https://www.jetbrains.com/lp/compose-mpp/) mostly because I'm not great with Css/Html/Js and because I like working with Kotlin/Compose/Coroutines

Open to hear feedback about the pros and cons of this approach

Screenshot:
ws

### CODE REVIEW

**Comment by feresr** [comment]
code in `libs/fgdbg/client/src/main/kotlin/Main.kt` is likely to change substantially. I'm not even sure we'll use sockets at this point. 
This PR is mostly to set up the client project structure, dependencies, etc. 
And to discuss if this is the right approach for this tool

**Comment by romainguy** [comment]
As the manager of the Compose team I can't be against it 😁

**Comment by feresr** [comment]
@romainguy haha happy to hear that!
Actively learning Compose, don't hold back on feedback 😀

File: libs/fgdbg/client/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import org.jetbrains.compose.compose
+import org.jetbrains.compose.desktop.application.dsl.TargetFormat
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+    kotlin("jvm") version "1.6.10"
+    id("org.jetbrains.compose") version "1.1.1"
+}
+
+group = "com.google.filament"
```
Comment (feresr) [inline libs/fgdbg/client/build.gradle.kts:10]: should I suffix`.fgdbg` here 🤔

File: libs/fgdbg/client/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+version = "1.0"
+val ktorVersion: String by project
+
+repositories {
+    google()
+    mavenCentral()
+    maven("https://maven.pkg.jetbrains.space/public/p/compose/dev")
+  maven {
```
Comment (romainguy) [inline libs/fgdbg/client/build.gradle.kts:18]: Indentation is incorrect on this maven block

File: libs/fgdbg/client/src/main/kotlin/Main.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import androidx.compose.material.MaterialTheme
```
Comment (romainguy) [inline libs/fgdbg/client/src/main/kotlin/Main.kt:1]: Add a copyright/license block

File: libs/fgdbg/client/src/main/kotlin/Main.kt [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import io.ktor.client.*
+import io.ktor.client.plugins.websocket.*
+import io.ktor.http.*
+import io.ktor.websocket.Frame
+import io.ktor.websocket.readText
+import kotlinx.coroutines.delay
+
+private enum class ConnectionState { UNKNOWN, CONNECTED, CONNECTING, ERROR }
+
+private data class Connection(
```
Comment (romainguy) [inline libs/fgdbg/client/src/main/kotlin/Main.kt:30]: Our code style uses 4 spaces instead of 2.

**Review by romainguy — COMMENTED** [review]
Just small nits

File: libs/fgdbg/client/src/main/kotlin/Main.kt (hunk 3 above)
Comment (romainguy) [inline libs/fgdbg/client/src/main/kotlin/Main.kt:1]: More important: tools should go in tools/, not in libs/

**Review by romainguy — COMMENTED** [review]
Added an important note. Also eventually we'll need a CMake target to build the tool.

File: libs/fgdbg/client/build.gradle.kts (hunk 1 above)
Comment (romainguy) [reply libs/fgdbg/client/build.gradle.kts:10]: No the group is correct. If we wanted to publish this we would differentiate it by using a separate artifact id containing `fgfbg`.

File: libs/fgdbg/client/build.gradle.kts (hunk 2 above)
Comment (feresr) [reply libs/fgdbg/client/build.gradle.kts:18]: good catch!

File: libs/fgdbg/client/src/main/kotlin/Main.kt (hunk 3 above)
Comment (feresr) [reply libs/fgdbg/client/src/main/kotlin/Main.kt:17]: added

Comment (feresr) [reply libs/fgdbg/client/src/main/kotlin/Main.kt:17]: Makes sense, I'll do this on a followup PR to keep this one in scope. 
Should I move everything or just `fgdbg/client`?
I am mostly mirroring what `matdbg` does, should `matdbg/web` go inside `tools/` too?

**Comment by feresr** [comment]
> we'll need a CMake target to build the tool.

This should be interesting. Do you mean have CMake call gradle under the hood?
Or can compose projects be built with cmake too?

**Review by bejado — APPROVED** [review]
Excited for this 😎

**Pull request merged**